Doc/library/asyncio-task.rst

```diff
@@ -527,6 +527,41 @@ Running Tasks Concurrently
       and there is no running event loop.
 
 
+Eager Task Factory
+==================
+
+.. function:: eager_task_factory(loop, coro, *, name=None, context=None)
+
+    A task factory for eager task execution.
+
+    When using this factory (via ``loop.set_task_factory(asyncio.eager_task_factory)``),
+    coroutines that are able to complete synchronously (without blocking)
+    are returned immediately as a finished :class:`Task`.
+
+    This task factory attempts to execute the coroutine ``coro`` immediately
+    (before scheduling the task to the event loop), until it either blocks,
+    returns, or raises.
+    If the coroutine returns or raises, a finished :class:`Task` is returned,
+    and the task is never scheduled to the event loop. If the coroutine blocks,
+    the (pending) :class:`Task` is scheduled and returned.
```

Comment:
```suggestion
    When using this factory (via ``loop.set_task_factory(asyncio.eager_task_factory)``),
    coroutines begin execution synchronously on `Task` construction. `Task`s are only
    scheduled  on the event loop if they block. This can be a performance improvement as
    the overhead  of loop scheduling is avoided for coroutines that complete synchronously.
    A common  example where this is beneficial is coroutines which employ caching to avoid
    actual I/O where possible.
```